### PR TITLE
add image regression task dataset format

### PR DESCRIPTION
**Pull request by juliagongms**

### CODE REVIEW

File: COCO_DATA_FORMAT.md [hunk 1]
```diff
@@ -94,3 +94,18 @@ Specifically, **only** image files are supported for the label files. The ground
     ]
 }
 ```
+
+## Regression
+
+Here is one example of the json file for regression task, where the "target" in the "annotations" field is a real-valued number (e.g. a score, an age, etc.). Note that each image should only have one regression target (i.e. there should be exactly one annotation for each image).
```
Comment (shonohs) [inline COCO_DATA_FORMAT.md]: In the foreseeable future, we don't have a plan to use a dataset with multiple targets per image?

Comment (cy-bai) [inline COCO_DATA_FORMAT.md:100]: Are we going to have multi-target regression datasets? If so, we could make it more general

Comment (cy-bai) [reply COCO_DATA_FORMAT.md]: +1

Comment (pj-ms) [reply COCO_DATA_FORMAT.md]: @shonohs @cy-bai we can use multitask dataset to organize the multiple target data, where multiple coco will be provided.

File: COCO_DATA_FORMAT.md [hunk 2]
```diff
@@ -94,3 +94,18 @@ Specifically, **only** image files are supported for the label files. The ground
     ]
 }
 ```
+
```
Comment (pj-ms) [inline COCO_DATA_FORMAT.md:97]: The code in this repo require update as well to be able to consume this new format of data, with unit tests

Comment (pj-ms) [reply COCO_DATA_FORMAT.md:97]: along with the readme file

File: COCO_DATA_FORMAT.md (hunk 1 above)
Comment (juliagongms) [reply COCO_DATA_FORMAT.md]: Sounds good; we can use the multitask dataset format with multiple COCO files if there are multiple targets.

File: COCO_DATA_FORMAT.md (hunk 2 above)
Comment (juliagongms) [reply COCO_DATA_FORMAT.md:97]: Made the corresponding changes, thanks!

**Review by cy-bai — APPROVED** [review]

File: vision_datasets/common/constants.py [hunk 3]
```diff
@@ -6,7 +6,8 @@ class DatasetTypes:
     IMCAP = 'image_caption'
     IMAGE_TEXT_MATCHING = 'image_text_matching'
     IMAGE_MATTING = 'image_matting'
-    VALID_TYPES = [IC_MULTILABEL, IC_MULTICLASS, OD, MULTITASK, IMCAP, IMAGE_TEXT_MATCHING, IMAGE_MATTING]
+    REGRESSION = 'regression'
+    VALID_TYPES = [IC_MULTILABEL, IC_MULTICLASS, OD, MULTITASK, IMCAP, IMAGE_TEXT_MATCHING, IMAGE_MATTING, REGRESSION]
```
Comment (pj-ms) [inline vision_datasets/common/constants.py]: As we have text data now, can we make it IMAGE_REGRESSION = 'image_regression' to be more clear? Thanks

File: vision_datasets/common/data_manifest.py [hunk 4]
```diff
@@ -101,7 +101,8 @@ def __init__(self, id, img_path, width, height, labels, label_file_paths=None, l
                 image_caption: [caption1, caption2, ...];
                 image_text_matching: [(text1, match (0 or 1), text2, match (0 or 1), ...)];
                 multitask: dict[task, labels];
-                image_matting: [mask1, mask2, ...], each mask is a 2D numpy array that has the same width and height with the image.
+                image_matting: [mask1, mask2, ...], each mask is a 2D numpy array that has the same width and height with the image;
+                regression: [target1, target2, ...].
```
Comment (pj-ms) [inline vision_datasets/common/data_manifest.py]: Should this be a single float?

File: vision_datasets/common/constants.py (hunk 3 above)
Comment (juliagongms) [reply vision_datasets/common/constants.py]: Changed globally. Thanks!

File: vision_datasets/common/data_manifest.py (hunk 4 above)
Comment (juliagongms) [reply vision_datasets/common/data_manifest.py]: Yes; made the change. Thanks!

**Review by pj-ms — APPROVED** [review]

**Pull request merged**